sdk/cloudmachine/Azure.CloudMachine/src/CoreServices/StorageServices.cs

```diff
@@ -231,4 +273,17 @@ public void WhenBlobUploaded(Action<StorageFile> function)
 #pragma warning restore AZC0102 // Do not use GetAwaiter().GetResult().
 
     }
+
+    /// <summary>
+    /// Adds a function to be called when a blob is uploaded.
+    /// </summary>
+    /// <param name="function"></param>
+    public void WhenUploaded(Action<BinaryData> function)
```

Comment:
Do we have a scenario in mind for this method? It would mean that the provided function has no information about the file being uploaded.

Reply:
Yeah, I use it in the demo app.